camel/agents/task_agent.py

```diff
@@ -181,3 +183,192 @@ def run(
 
         sub_tasks_msg = task_response.msgs[0]
         return TextPrompt(sub_tasks_msg.content)
+
+
+class TaskCreationAgent(ChatAgent):
+    r"""An agent that helps create new tasks based on the objective
+    and last completed task. Compared to :obj:`TaskPlannerAgent`,
+    it's still a task planner, but it has more context information
+    like last task and incompleted task list.
+
+    Attributes:
+        task_creator_prompt (TextPrompt): A prompt for the agent to create
+            new tasks.
+
+    Args:
+        model (ModelType, optional): The type of model to use for the agent.
+            (default: :obj:`ModelType.GPT_3_5_TURBO`)
+        model_config (Any, optional): The configuration for the model.
+            (default: :obj:`None`)
+        output_language (str, optional): The language to be output by the
+            agent. (default: :obj:`None`)
+        objective (str): The objective of the Agent to perform the task.
+    """
+
+    def __init__(
+        self,
+        model: Optional[ModelType] = None,
+        model_config: Optional[Any] = None,
+        output_language: Optional[str] = None,
+        objective: str = "",
```

Comment:
Let's put `objective` as a required positional argument.
```suggestion
        objective: Union[str, TextPrompt],
        model: Optional[ModelType] = None,
        model_config: Optional[Any] = None,
        output_language: Optional[str] = None,
```